docs/design/values.md

```diff
@@ -629,6 +773,233 @@ The model of initialization of returns also facilitates the use of
+-   The outcome of an expression with struct form is a struct of outcomes.
+
+An expression and its outcome always have the same form.
+
+### Form conversions
+
+A conversion between forms can be broken down into up to three steps: type
+conversion, category conversion, and phase conversion. These convert the form to
+a particular target type, category, and phase component (respectively). These
+steps aren't fully orthogonal: type conversions can change the category and
+phase components as a byproduct, and category conversions can change the phase
+component. However, category conversions can't change the type component, and
+phase conversions can't change either of the other two, so converting the type,
+then category, then phase, ensures that we converge on the desired result.
+
+Any of these steps may be omitted, depending on whether the context imposes
+requirements on the corresponding component. Most commonly, an operand position
+requires its operand to have a primitive form with a particular category,
+usually with a particular type, and sometimes with a particular phase.
+
+In some cases an expression's outcome is _discarded_, such as when the
+expression is used as a statement, or is matched with an
+[unused binding pattern](pattern_matching.md#unused-bindings). Discarding an
+outcome is a form conversion that does nothing except materialize any
+initializing sub-outcomes, in order to satisfy the requirement that every
+initializing outcome is materialized.
+
+Phase conversions are straightforward, because they cannot change the form
+structure; they can only apply primitive phase conversions to primitive
+sub-forms. Type and category conversions are more complex, and are covered in
+the next two sections.
+
+#### Type conversions
+
+See [here](expressions/implicit_conversions.md) for overall information about
+type conversions. Conversions involving struct, tuple, and array types are
+described here because of their unique interactions with expression forms.
+
+> **TODO:** A forthcoming proposal is expected to update the type conversion
+> interfaces to permit user-defined conversions to depend on the form of the
+> input, and customize the form of the output. Once that is done, these "built
+> in" conversions should be presented as implementations of those interfaces,
+> possibly with some "magic" for things like introspecting on struct field
+> names.
+
+Each of the conversions described in this section is explicit if and only if it
+invokes another explicit type conversion. Otherwise, it is implicit.
+
+An outcome `source` that has a struct type can be converted to a struct type
+`Dest` if they have the same set of field names:
+
+-   If the type of `source` is `Dest`, return `source`.
+-   If `source` is a struct outcome, for each field name `F` in `Dest`, in
+    `Dest`'s field order, type-convert `source.F` to `Dest.F`. Return a struct
+    outcome where each field `F` is set to the outcome of the corresponding
+    conversion.
```

Comment:
The toolchain produces a primitive form in this case. Does the difference matter?